registries/zookeeper/resolver/resolver.go

```diff
@@ -105,3 +108,22 @@ func (z *zookeeperResolver) Diff(cacheKey string, prev, next discovery.Result) (
 func (z *zookeeperResolver) Name() string {
 	return z.uniqueName
 }
+
```

Comment:
This function is hard to understand, please add some comment and an example input/output

Reply:
Also better covered with unit tests

Reply:
Ack.